core/templates/domain/collection/collection-update.service.spec.ts

```diff
@@ -16,79 +16,94 @@
-
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    $provide.value('ChangeObjectFactory', Change);
-  }));
-
-  beforeEach(angular.mock.inject(function($injector) {
-    CollectionUpdateService = $injector.get('CollectionUpdateService');
-    UndoRedoService = $injector.get('UndoRedoService');
-
-    const sampleCollectionBackendObject = {
+import { TestBed } from '@angular/core/testing';
+
+import { Collection, CollectionBackendDict } from 'domain/collection/collection.model';
+import { CollectionUpdateService } from 'domain/collection/collection-update.service';
+import { UndoRedoService } from 'domain/editor/undo_redo/undo-redo.service';
+import { LearnerExplorationSummaryBackendDict } from 'domain/summary/learner-exploration-summary.model';
+
+describe('Collection update service', () => {
+  let collectionUpdateService: CollectionUpdateService = null;
+  let undoRedoService: UndoRedoService = null;
+  let learnerExplorationSummaryBackendDict:
+    LearnerExplorationSummaryBackendDict = null;
+  let _sampleCollection = null;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: []
+    });
+    collectionUpdateService = TestBed.get(CollectionUpdateService);
+    undoRedoService = TestBed.get(UndoRedoService);
+  });
+
+  beforeEach(() => {
+    learnerExplorationSummaryBackendDict = {
+      activity_type: 'exploration',
+      category: 'a category',
+      community_owned: false,
+      created_on_msec: 1591296635736.666,
+      id: 'collection_id',
+      last_updated_msec: 1591296737470.528,
+      ratings: {
+        1: 0,
+        2: 0,
+        3: 0,
+        4: 0,
+        5: 0
+      },
+      human_readable_contributors_summary: {},
+      language_code: 'en',
+      num_views: 0,
+      objective: 'Test Objective',
+      status: 'public',
+      tags: [],
+      thumbnail_bg_color: '#cd672b',
+      thumbnail_icon_url: '/subjects/Algebra.svg',
+      title: 'a title'
+    };
+    let sampleCollectionBackendObject: CollectionBackendDict = {
       id: 'collection_id',
       title: 'a title',
       objective: 'an objective',
       language_code: 'en',
       tags: [],
       category: 'a category',
-      version: '1',
+      version: 1,
+      schema_version: 1,
       nodes: [{
         exploration_id: 'exp_id0',
-        exploration: {}
+        exploration_summary: learnerExplorationSummaryBackendDict
       }],
       playthrough_dict: {
         next_exploration_id: 'expId',
         completed_exploration_ids: ['expId2']
       }
     };
     _sampleCollection = Collection.create(
-      // TODO(#10875): Fix type mismatch.
-      sampleCollectionBackendObject as unknown as CollectionBackendDict);
-  }));
+      sampleCollectionBackendObject);
+  });
 
   it('should add/remove a new collection node to/from a collection',
-    function() {
+    () => {
       expect(_sampleCollection.getExplorationIds()).toEqual(['exp_id0']);
```

Comment:
should we test the private functions like this? we should use the helper functions to test these private functions.

Reply:
`getExplorationIds` in `Collection` is not a private function.Which private function did you intend? @darksun27

Reply:
Ohh so sorry. I misread the underscore.